core/src/test/java/tech/tablesaw/TableTest.java

```diff
@@ -64,4 +71,99 @@ public void testRowCount() throws Exception {
         floatColumn.append(2.2342f);
         assertEquals(2, table.rowCount());
     }
+
+    @Test
+    public void testAppend() throws Exception {
+        int appendSize = random.nextInt(ROWS_BOUNDARY);
+        appendTableWithSingleColumn(appendSize);
+    }
+
+    @Test
+    public void testAppendEmptyTable() throws Exception {
+        appendTableWithSingleColumn(0);
+    }
+
+    @Test
+    public void testAppendToNonEmptyTable() throws Exception {
+        int appendSize = random.nextInt(ROWS_BOUNDARY);
+        populateColumn(floatColumn);
+        assumeFalse(table.isEmpty());
+        appendTableWithSingleColumn(appendSize);
+    }
+
+    @Test
+    public void testAppendEmptyTableToNonEmptyTable() throws Exception {
+        populateColumn(floatColumn);
+        assumeFalse(table.isEmpty());
+        appendTableWithSingleColumn(0);
+    }
+
+    @Test
+    public void testAppendMultipleColumns() throws Exception {
+        FloatColumn column = new FloatColumn("e1");
+        table.addColumn(column);
+        FloatColumn first = floatColumn.emptyCopy();
+        FloatColumn second = column.emptyCopy();
+        int firstColumnSize = populateColumn(first);
+        int secondColumnSize = populateColumn(second);
+        Table tableToAppend = Table.create("populated", first, second);
+        table.append(tableToAppend);
+        assertColumnSize(floatColumn, firstColumnSize);
+        assertColumnSize(column, secondColumnSize);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testAppendNull() throws Exception {
+        table.append(null);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testAppendTableWithNonExistingColumns() throws Exception {
+        Table tableToAppend = Table.create("wrong", doubleColumn);
+        table.append(tableToAppend);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testAppendTableWithAnotherColumnName() throws Exception {
+        FloatColumn column = new FloatColumn("42");
+        Table tableToAppend = Table.create("wrong", column);
+        table.append(tableToAppend);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testAppendTableWithDifferentShape() throws Exception {
+        FloatColumn column = new FloatColumn("e1");
+        table.addColumn(column);
+        Table tableToAppend = Table.create("different", column);
+        assumeTrue(table.columns().size() == 2);
```

Comment:
It seems like all the places where assume is used we should use assert instead. I would expect the test to fail if this is not true rather than be ignored